shared/constants/wallets.js

```diff
@@ -1,14 +1,133 @@
 // @flow
 import * as I from 'immutable'
 import * as Types from './types/wallets'
+import * as RPCTypes from './types/rpc-stellar-gen'
+import {invert} from 'lodash'
+
+const balanceDeltaToString = invert(RPCTypes.localBalanceDelta)
+const statusSimplifiedToString = invert(RPCTypes.localPaymentStatus)
 
 const makeReserve: I.RecordFactory<Types._Reserve> = I.Record({
   amount: '',
   description: '',
 })
 
 const makeState: I.RecordFactory<Types._State> = I.Record({
-  hello: 'world',
+  assetsMap: I.Map(),
+  paymentsMap: I.Map(),
+  walletMap: I.Map(),
+})
+
+const makeWallet: I.RecordFactory<Types._Wallet> = I.Record({
+  accountID: '',
+  balanceDescription: '',
+  isDefault: false,
+  name: '',
+})
+
+const walletResultToWallet = (w: RPCTypes.WalletAccountLocal) => {
+  const {accountID, balanceDescription, isDefault, name} = w
+  return makeWallet({
+    accountID,
+    balanceDescription,
+    isDefault,
+    name,
+  })
+}
+
+const makeAssets: I.RecordFactory<Types._Assets> = I.Record({
+  assetCode: '',
+  balanceAvailableToSend: '',
+  balanceTotal: '',
+  issuer: '',
+  name: '',
+  worth: '',
+  worthCurrency: '',
 })
 
-export {makeReserve, makeState}
+const assetsResultToAssets = (w: RPCTypes.AccountAssetLocal) => {
+  const {assetCode, balanceAvailableToSend, balanceTotal, issuer, name, worth, worthCurrency} = w
+  return makeAssets({
+    assetCode,
+    balanceAvailableToSend,
+    balanceTotal,
+    issuer,
+    name,
+    worth,
+    worthCurrency,
+  })
+}
+
+const makePayment: I.RecordFactory<Types._Payment> = I.Record({
+  amountDescription: '',
+  delta: 'none',
+  error: '',
+  id: '',
+  note: '',
+  noteErr: '',
+  source: '',
+  sourceType: '',
+  statusDescription: '',
+  statusDetail: '',
+  statusSimplified: 'none',
+  target: '',
+  targetType: '',
+  time: 0,
+  worth: '',
+  worthCurrency: '',
+})
+
+const paymentResultToPayment = (w: RPCTypes.PaymentOrErrorLocal) => {
+  if (!w) {
+    return makePayment({error: 'No payments returned'})
+  }
+  if (!w.payment) {
+    return makePayment({error: w.err})
+  }
+  const {
+    amountDescription,
+    delta,
+    id,
+    note,
+    noteErr,
+    source,
+    sourceType,
+    statusDescription,
+    statusDetail,
+    statusSimplified,
+    target,
+    targetType,
+    time,
+    worth,
+    worthCurrency,
+  } = w.payment
```

Comment:
i don't think this kind of destructuring very readable, like it helps makes the lines below less wide but adds 15 lines to the file. if we did
`const p = w.payment` then below

```
id: p.id,
note: p.note,
``` 
etc that doesn't seem so bad to me.

You don't have to change this as its a style thing but i find if the file is longer lines wise its just harder to consume